### DISCUSSION TITLE
How do you keep Vercel’s "Execution" usage under control?

### DISCUSSION THREAD

**Discussion by miguderp** [vercel/next.js#33851] [Help] [answered]
Hi everyone 👋🏻 

Sorry this is going to be a long text, hope some people will find the will to read it all and help us 🙏🏻 

We have recently launched a website for an online magazine client that gets a few visitors (15/30 at any given time, peaks at 80) and got a few bad surprises with our data usage on our Pro team plan.

Since it's a news website, a lot of navigation items and listings on the homepage caused us a lot of Execution time for a couple days before it caught our attention.

![image](https://user-images.githubusercontent.com/7307687/151945955-a587c78f-6257-4b55-9cf6-69a1442dd1b3.png)

Since then, we have been forced to do a few things in order to try to keep things under control:

1. Swap hyperlinks to `<Link prefetch={false} />` so it doesn't try to revalidate everything just by scrolling through the homepage
2. We initially had [SWR](https://swr.vercel.app/) on every page so it would refresh the content on the client-side if the revalidate was still waiting to be renewed. We were trying to be technically fancy but that's entirely removed now.
3. Increase our `revalidate` to `360` (is that enough? How do you guys set that value at for a website that has a lot of dynamic content?)
4. Since it's a heavy website on images too, we swapped our images to `<Image unoptimized={true} />` because we actually didn't really need it since images were optimized at the source (headless WordPress). I did not find any confirmation anywhere but is it wrong to believe that Next's image component uses serverless functions to optimize the images? Hence disabling it would positively affect the Execution time?

Since then (we have yet to see the impact of point 4), it's a lot better but even then we still seem to get between 30 and 50gb on a daily basis which by itself is too much for a single project (nearing 1TB monthly by itself alone… we have other client projects on our account!).

We were considering these last few options (please tell me if there are more available):

1. Reducing the memory allocated to functions by half [as underlined here](https://vercel.com/docs/cli#project-configuration/functions). Would that be a bad idea? Would all my functions timeout from the moment it gets lowered?
2. Find another hosting provider as Vercel's hosting choices between pro and enterprise is too steep for us. Maybe Netlify? Their pricing plan seems to state that "Serverless functions" are unlimited with their Business account. Would "Execution" from Vercel enter in that category?

We know there's an Enterprise plan but the price change is way too steep for us, starting at 36k/yearly apparently.
We have also contacted Vercel's support but unfortunately it does not seem that purchasing "packs" of 1TB/h is possible under Pro accounts, only Enterprise.

For guys that have websites with high affluence that need regular updates, what are your tricks to reduce the usage?
Do you just accept the fact to have revalidates really high and educate your client about how to live with it?

Unfortunately it seems that Vercel has been working on an API to flush specific pages built with ISR but [this discussion seems a bit stale for almost a year](https://github.com/vercel/next.js/discussions/11552#discussioncomment-646963)…

Thank you for reading!

**Comment by wwwdepot** [comment, marked as answer]
How can I reduce my Serverless Execution usage on Vercel? https://vercel.com/support/articles/how-can-i-reduce-my-serverless-execution-usage-on-vercel

In short: make functions use less memory and faster.

Anyway, you might want to think about the last paragraph from the support article.
